### PR TITLE
[WIP] feat!(detector): use vuls2 for redhat/alma/rocky

### PR DESCRIPTION
**Pull request by shino**
If this Pull Request is work in progress, Add a prefix of “[WIP]” in the title.

# What did you implement:

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:
You don't have to satisfy all of the following.

- [ ] Write tests
- [ ] Write documentation
- [ ] Check that there aren't other open pull requests for the same issue/feature
- [ ] Format your source code by `make fmt`
- [ ] Pass the test by `make test`
- [ ] Provide verification config / commands
- [ ] Enable "Allow edits from maintainers" for this PR
- [ ] Update the messages below

***Is this ready for review?:*** NO  

# Reference

* https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/

### CODE REVIEW

File: detector/vuls2/vuls2.go [hunk 1]
```diff
@@ -0,0 +1,633 @@
...
+			}
+			return ps
+		}(),
+		CPE: func() []string {
+			s, ok := old.Config.Scan.Servers[old.ServerName]
+			if !ok {
+				return nil
+			}
+			return s.CpeNames
+		}(),
```
Comment (MaineK00n) [inline detector/vuls2/vuls2.go:153]: CPE detection is not performed on the nightly side, so it does not seem necessary.
```suggestion
```

File: detector/vuls2/vuls2.go [hunk 2]
```diff
@@ -0,0 +1,633 @@
...
+		}
+		for rootID, d := range m {
+			base, ok := detected[rootID]
+			if !ok {
+				base = detectTypes.VulnerabilityData{ID: rootID}
+			}
+			base.Detections = append(base.Detections, d)
+			detected[rootID] = base
+		}
+	}
```
Comment (MaineK00n) [inline detector/vuls2/vuls2.go:192]: CPE detection is not performed on the nightly side, so it does not seem necessary.
```suggestion
```

File: detector/vuls2/vuls2.go [hunk 3]
```diff
@@ -0,0 +1,633 @@
...
+			Version:        old.RunningKernel.Version,
+			RebootRequired: old.RunningKernel.RebootRequired,
+		},
+		OSPackages: func() []scanTypes.OSPackage {
+			ps := make([]scanTypes.OSPackage, 0, len(pkgs))
+			for _, p := range pkgs {
+				ps = append(ps, p)
+			}
+			return ps
+		}(),
```
Comment (MaineK00n) [inline detector/vuls2/vuls2.go:146]:
```suggestion
		OSPackages: slices.Collect(maps.Values(pkgs)),
```

File: detector/vuls2/db.go [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+func (c Config) New() (db.DB, error) {
+	vuls2Config := db.Config{
+		Type: "boltdb",
+		Path: c.Path,
+	}
+
+	dbc, err := vuls2Config.New()
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to new vuls2 db. err; %w", err)
```
Comment (MaineK00n) [inline detector/vuls2/db.go:81]:
```suggestion
		return nil, xerrors.Errorf("Failed to new vuls2 db. err: %w", err)
```

File: detector/vuls2/db.go [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+	Path       string
+	SkipUpdate bool
+	Quiet      bool
+}
+
+func (c Config) Refresh() error {
+
+	lastModified, found, err := c.loadLastModified()
+	if err != nil {
+		return xerrors.Errorf("Failed to load vuls2 db metadata. err; %w", err)
```
Comment (MaineK00n) [inline detector/vuls2/db.go:51]:
```suggestion
		return xerrors.Errorf("Failed to load vuls2 db metadata. err: %w", err)
```

File: detector/vuls2/db.go [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+	SchemaVersion = 0
+	dbMediaType   = "application/vnd.vulsio.vuls.db.layer.v1+zstd"
+)
+
+var (
+	// DefaultGHCRRepository is GitHub Container Registry for vuls2 db for vuls0
+	DefaultGHCRRepository = fmt.Sprintf("%s:%d", "ghcr.io/vulsio/vuls-nightly-db", SchemaVersion)
+
+	// DefaultPath is the path for vuls2 db file
+	DefaultPath = filepath.Join(utilos.UserCacheDir(), "vuls.db")
```
Comment (shino) [inline detector/vuls2/db.go:37]: Set default path to CWD/vuls.db

File: config/config.go [hunk 7]
```diff
@@ -45,6 +45,7 @@ type Config struct {
 	Metasploit MetasploitConf `json:"metasploit,omitempty"`
 	KEVuln     KEVulnConf     `json:"kevuln,omitempty"`
 	Cti        CtiConf        `json:"cti,omitempty"`
+	Vuls2      Vuls2DictConf  `json:"vuls2Dict,omitempty"`
```
Comment (shino) [inline config/config.go:48]: Moooore better name!!!

File: detector/vuls2/db.go (hunk 6 above)
Comment (shino) [reply detector/vuls2/db.go]: https://github.com/future-architect/vuls/compare/5c54a7d8db4e217a887e72462d3d0016a747458e..15866221afd8e9ecfe42b36392ed2296ddd70f23

File: config/config.go (hunk 7 above)
Comment (shino) [reply config/config.go:50]: Hmm,,, difficult thing

**Pull request closed without merging**